packages/schematics/angular/library-entrypoint/index_spec.ts

```diff
@@ -0,0 +1,164 @@
+      workspaceTree,
+    );
+    const content = tree.readContent('/projects/foo/src/lib/foo-secondary/README.md');
+    expect(content).toMatch('# foo/foo-secondary');
+  });
+
+  it('should set a custom entryfile', async () => {
+    workspaceTree = await schematicRunner.runSchematic('library', libaryOptions, workspaceTree);
+    const tree = await schematicRunner.runSchematic(
+      'secondary',
+      { ...defaultOptions, entryFile: 'my-index' },
+      workspaceTree,
+    );
+    const files = tree.files;
+    expect(files).toEqual(
+      jasmine.arrayContaining([
+        '/projects/foo/src/lib/foo-secondary/README.md',
+        '/projects/foo/src/lib/foo-secondary/ng-package.json',
+        '/projects/foo/src/lib/foo-secondary/src/my-index.ts',
+      ]),
+    );
+  });
+
+  it('should handle scope packages', async () => {
+    workspaceTree = await schematicRunner.runSchematic(
+      'library',
+      { ...libaryOptions, name: '@scope/package' },
+      workspaceTree,
+    );
+    const tree = await schematicRunner.runSchematic(
+      'secondary',
+      { ...defaultOptions, name: 'testing', project: '@scope/package' },
+      workspaceTree,
+    );
+    const files = tree.files;
+    expect(files).toEqual(
+      jasmine.arrayContaining([
+        '/projects/scope/package/src/lib/testing/README.md',
+        '/projects/scope/package/src/lib/testing/ng-package.json',
+        '/projects/scope/package/src/lib/testing/src/public-api.ts',
+      ]),
+    );
+  });
+
+  it(`should add paths mapping to the tsconfig`, async () => {
+    workspaceTree = await schematicRunner.runSchematic(
+      'library',
+      { ...libaryOptions, name: '@scope/package' },
+      workspaceTree,
+    );
+    const tree = await schematicRunner.runSchematic(
+      'secondary',
+      { ...defaultOptions, name: 'testing', project: '@scope/package' },
+      workspaceTree,
+    );
+
+    const tsConfigJson = getJsonFileContent(tree, 'tsconfig.json');
+    expect(tsConfigJson.compilerOptions.paths['@scope/package/testing']).toEqual([
+      './dist/scope/package/testing',
+    ]);
+  });
+
+  it(`should append to existing paths mappings`, async () => {
+    workspaceTree = await schematicRunner.runSchematic(
+      'library',
+      { ...libaryOptions, name: '@scope/package' },
+      workspaceTree,
+    );
+    workspaceTree.overwrite(
+      'tsconfig.json',
+      JSON.stringify({
+        compilerOptions: {
+          paths: {
+            'unrelated': ['./something/else.ts'],
+            '@scope/package/testing': ['libs/*'],
+          },
+        },
+      }),
+    );
+    const tree = await schematicRunner.runSchematic(
+      'secondary',
+      { ...defaultOptions, name: 'testing', project: '@scope/package' },
+      workspaceTree,
+    );
+
+    const tsConfigJson = getJsonFileContent(tree, 'tsconfig.json');
+    expect(tsConfigJson.compilerOptions.paths['@scope/package/testing']).toEqual([
+      'libs/*',
+      './dist/scope/package/testing',
```

Comment:
This is incorrect because, in this scenario, dist should take precedence.

Reply:
Hi @alan-agius4 I hope I fixed it correctly otherwise I misunderstood your comment.